src/Features/Core/Portable/Snippets/Snippet.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    /// <summary>
+    /// Encapsulates the information that makes up a Snippet.
+    /// What is used to create the CompletionChange in the 
+    /// AbstractSnippetCompletionProvider.
+    /// </summary>
+    internal readonly struct Snippet
+    {
+        public readonly string SnippetType;
+        public readonly TextChange TextChange;
```

Comment:
what if a snippet needs to make many text changes?

Reply:
Things like an IDisposable snippet might insert fields, methods and the like. Those would be in different locations.